### DISCUSSION TITLE
Not quite a bug: function without an environment causes failure

### DISCUSSION THREAD

**Discussion by billdenney** [ropensci/targets#878] [Ideas]
This is one of those "why in the world would you do that, Bill" kinds of issues.  I almost posted it as a bug, but it didn't quite feel like a bug because it is rather off-the-wall.

I accidentally found an issue where when I assigned `attr<-` to a different name to make it a bit easier to use (mainly, easier to type), it caused `targets` to fail.  I think that this is an "ok, so don't do that" kind of usage error, but I wanted to let you know and to let others know if they hit it.  Maybe there is some kind of simple fix where it could be considered a feature request.  Or maybe it's a bug in R.

The issue relates to this line:

```r
set_attr <- `attr<-`
```

``` r
library(targets)
#> Warning: package 'targets' was built under R version 4.1.3
tar_script({
    library(targets)
    set_attr <- `attr<-`
    foo <- 1
    list(tar_target(x, set_attr(foo, "bar", "baz")))
})
tar_make()
#> Error in as.function.default(c(as.list(formals(fun)), list(value)), envir) : 
#>   use of NULL environment is defunct
#> In addition: Warning message:
#> package 'targets' was built under R version 4.1.3 
#> Error in `tar_throw_run()`:
#> ! callr subprocess failed: use of NULL environment is defunct
#> Visit https://books.ropensci.org/targets/debugging.html for debugging advice.
```

<sup>Created on 2022-07-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by wlandau** [comment]
Yeah, overwriting functions in {base} can throw off any number of functions in any package. I would suggest writing your own wrapper function that calls attr() rather than overwriting attr() itself.
